pennylane/operation.py

```diff
@@ -297,7 +297,7 @@ class WiresEnum(IntEnum):
     an operation acts on"""
 
     AnyWires = -1
-    AllWires = 0
+    AllWires = -2
```

Comment:
Any reason why we need to use an `IntEnum` at all? Why not just something distinctly different, like a string?

Reply:
no strong reason, just doing a minimal change to something that `len(wires)` won't return. we can make it anything!

AllWires isn't really used anyway. Perhaps it's time to scrap this enum altogether. We really just need a way for an operator to say "I can have any number of wires!". The only remaining candidate for `AllWires` is Snapshot afaik